core/src/commands/publish_datapoint.rs

```diff
@@ -0,0 +1,420 @@
+            new_datapoint,
+            oracle_token_id,
+            reward_token_id,
+            public_key,
+        ),
+    }
+}
+
+pub fn build_subsequent_publish_datapoint_action(
+    pool_box_source: &dyn PoolBoxSource,
+    local_datapoint_box_source: &dyn LocalDatapointBoxSource,
+    wallet: &dyn WalletDataSource,
+    height: u32,
+    change_address: Address,
+    new_datapoint: i64,
+) -> Result<PublishDataPointAction, PublishDatapointActionError> {
+    let in_pool_box = pool_box_source.get_pool_box()?;
+    let in_oracle_box = local_datapoint_box_source.get_local_oracle_datapoint_box()?;
+    if *in_oracle_box.reward_token().amount.as_u64() == 0 {
+        return Err(PublishDatapointActionError::NoRewardTokenInOracleBox);
+    }
+
+    // Build the single output box
+    let mut builder = ErgoBoxCandidateBuilder::new(
+        in_oracle_box.get_box().value,
+        in_oracle_box.get_box().ergo_tree.clone(),
+        height,
+    );
+    let new_epoch_counter: i32 = (in_pool_box.epoch_counter() + 1) as i32;
+    builder.set_register_value(R4, in_oracle_box.public_key().into());
+    builder.set_register_value(R5, new_epoch_counter.into());
+    builder.set_register_value(
+        R6,
+        compute_new_datapoint(new_datapoint, in_oracle_box.rate() as i64).into(),
+    );
+    builder.add_token(in_oracle_box.oracle_token().clone());
+    builder.add_token(in_oracle_box.reward_token().clone());
+    let output_candidate = builder.build()?;
+
+    let unspent_boxes = wallet.get_unspent_wallet_boxes()?;
+    let tx_fee = BoxValue::SAFE_USER_MIN;
+    let box_selector = SimpleBoxSelector::new();
+    let selection = box_selector.select(unspent_boxes, tx_fee, &[])?;
+    let mut input_boxes = vec![in_oracle_box.get_box()];
+    input_boxes.append(selection.boxes.as_vec().clone().as_mut());
+    let box_selection = BoxSelection {
+        boxes: input_boxes.try_into().unwrap(),
+        change_boxes: selection.change_boxes,
+    };
+    let mut tx_builder = TxBuilder::new(
+        box_selection,
+        vec![output_candidate],
+        height,
+        tx_fee,
+        change_address,
+        BoxValue::MIN,
+    );
+
+    // The following context value ensures that `outIndex` in the oracle contract is properly set.
+    let ctx_ext = ContextExtension {
+        values: vec![(0, 0i32.into())].into_iter().collect(),
+    };
+    tx_builder.set_context_extension(in_oracle_box.get_box().box_id(), ctx_ext);
+    let tx = tx_builder.build()?;
+    Ok(PublishDataPointAction { tx })
+}
+
+pub fn build_publish_first_datapoint_action(
+    wallet: &dyn WalletDataSource,
+    height: u32,
+    change_address: Address,
+    new_datapoint: i64,
+    oracle_token_id: TokenId,
+    reward_token_id: TokenId,
+    public_key: ProveDlog,
+) -> Result<PublishDataPointAction, PublishDatapointActionError> {
+    // Build the single output box
+    let mut builder = ErgoBoxCandidateBuilder::new(
+        BoxValue::SAFE_USER_MIN,
```

Comment:
Just confirming if this a suitable value of ERG for the first oracle box?

Reply:
Yes. I don't see a reason to have any extra ERGs here so let's start with a minimal value.